### PR TITLE
fix(CameraControls): taking advantage on new released `.connect` method

### PR DESCRIPTION
**Pull request by abernier**
### Why / What

Now that [camera-controls](https://github.com/yomotsu/camera-controls/releases/tag/v1.38.0) lib implements `connect`/`disconnect` methods, I'm following [@drcmda recommendation](https://github.com/yomotsu/camera-controls/issues/262#issuecomment-1399431569) and update the `CameraControls` drei component to take advantage on

### Checklist

- [x] Documentation updated
- [x] Storybook entry added
- [x] Ready to be merged

### CODE REVIEW

File: src/core/CameraControls.tsx [hunk 1]
```diff
@@ -41,7 +41,8 @@ export const CameraControls = forwardRef<CameraControlsImpl, CameraControlsProps
   }, -1)
 
   useEffect(() => {
-    return () => void cameraControls.dispose()
+    cameraControls.connect(explDomElement)
+    return () => void cameraControls.disconnect()
```
Comment (yomotsu) [inline src/core/CameraControls.tsx:45]: @abernier I think we could keep `dispose()` at cleanup as same as other controls in drei. `disconnect()` may remain some features of CameraControls instance.
What do you think?
```suggestion
    return () => void cameraControls.dispose()
```

Comment (abernier) [reply src/core/CameraControls.tsx:45]: in my understanding, r3f will take care of calling `.dispose` thanks to https://docs.pmnd.rs/react-three-fiber/api/objects#disposal

Here it is more about "re-connecting" (`disconnect` + `connect`) once any useEffect deps changes

@drcmda or @CodyJasonBennett can you confirm?

Comment (CodyJasonBennett) [reply src/core/CameraControls.tsx:45]: Connect/disconnect should do and undo effectful stuff like listeners. I'd expect `dispose` to not be reversible, repeatable, or cheap to do at runtime.

Comment (abernier) [reply src/core/CameraControls.tsx:45]: Thank you @CodyJasonBennett, so in my mind this looks ok to me 👍🏻

Comment (yomotsu) [reply src/core/CameraControls.tsx:45]: @abernier 
That makes sense. Thank you for considering it.

**Pull request merged**